runtime/account/src/account.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright 2019-2020 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! The AccountId20 implementation.
+//!
+//! It includes AccountId conversations
+//!
+//! Ex:
+//! use std::str::FromStr;
+//! AccountId20::from_str("6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b").unwrap()
+
+use sp_core::{H160};
+use sp_std::hash::Hash;
+use codec::{Encode, Decode};
+#[cfg(feature = "std")]
+use sp_core::hexdisplay::HexDisplay;
+
+#[cfg(feature = "std")]
+use std::str::FromStr;
+
+#[cfg(feature = "full_crypto")]
+use sp_core::crypto::{PublicError};
+
+use pallet_evm::AddressMapping;
+
+/// An opaque 20-byte cryptographic identifier.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Default, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Hash))]
+pub struct AccountId20([u8; 20]);
```

Comment:
Is the plan to leave this in Moonbeam, or offer it up to Substrate? One good reason to contribute it to Substrate is that Jaco is generally willing to support types that are part of the Substrate repo.

If we do plan to contribute it to Substrate, I don't see any reason to wait. Opening a PR there will get review from a much wider range of devs.

Reply:
I don't have a strong opinion for it, but it doesn't seem others are interested in this feature.
The main reason I didn't include it was the complexity of adding stuff to Substrate when I started it